clang/lib/AST/DeclCXX.cpp

```diff
@@ -561,6 +561,42 @@ void CXXRecordDecl::addedClassSubobject(CXXRecordDecl *Subobj) {
     data().StructuralIfLiteral = false;
 }
 
+const CXXRecordDecl *CXXRecordDecl::getStandardLayoutBaseWithFields() const {
+#ifndef NDEBUG
+  {
+    assert(
+        isStandardLayout() &&
+        "getStandardLayoutBaseWithFields called on a non-standard-layout type");
+    unsigned NumberOfBasesWithFields = 0;
+    if (!field_empty())
+      ++NumberOfBasesWithFields;
+    std::set<const CXXRecordDecl *> UniqueBases;
```

Comment:
Have you considered `llvm::DenseSet`?

Reply:
@Endilll `llvm::SmallPtrSet` seemed more appropriate here